proxyd/server.go

```diff
@@ -674,6 +685,18 @@ func (s *Server) rateLimitSender(ctx context.Context, req *RPCReq) error {
 	return nil
 }
 
+func (s *Server) isAllowedChainId(chainId *big.Int) bool {
+	if s.allowedChainIds == nil || len(s.allowedChainIds) == 0 {
+		return true
+	}
+	for _, id := range s.allowedChainIds {
```

Comment:
[nit] most of the time `s.allowedChainIds` will be just one element, so we can save some CPU cycles by checking `if len(s.allowedChainIds) == 1 && s.allowedChainIds[0].Cmp(id) == 0` outside the for loop.

Reply:
the loop is short circuited, so it will return as soon as the first chain id matches
as we know, this array will always with one element or be super short